test/cruds/removing-sheet.spec.ts

```diff
@@ -109,6 +109,17 @@ describe('remove sheet', () => {
     expect(engine.sheetMapping.numberOfSheets()).toBe(0)
     expect(Array.from(engine.addressMapping.entries())).toEqual([])
   })
+
+  it('should remove a sheet with a cell reference to a value in the same sheet', () => {
```

Comment:
I think it could be a good idea to check if by removing one sheet, we're not affecting values in another sheet which is referenced. This does not have to be a scope of this PR, but maybe it's worth checking if it makes sense:

```javascript
  it('should not remove data from cell referenced in another sheet', () => {
    const engine = HyperFormula.buildFromSheets({
      Sheet1: [
        ['=Sheet2!A1)'],
      ],
      Sheet2: [
        ['1'],
      ],
    })

    engine.removeSheet(0)

    expect(Array.from(engine.sheetMapping.displayNames())).toEqual(['Sheet2'])
    expect(engine.getCellValue(adr('A1', 1))).toBe(1)
  })

  it('should not remove data from cell range referenced in another sheet', () => {
    const engine = HyperFormula.buildFromSheets({
      Sheet1: [
        ['=SUM(Sheet2!A1:A2)'],
      ],
      Sheet2: [
        ['1'],
        ['5'],
      ],
    })

    engine.removeSheet(0)

    expect(Array.from(engine.sheetMapping.displayNames())).toEqual(['Sheet2'])
    expect(engine.getCellValue(adr('A1', 1))).toBe(1)
    expect(engine.getCellValue(adr('A2', 1))).toBe(5)
  })
```

Let me know what you think.

Reply:
Sure, these test cases are absolutely valid. I added an `it` that covers both of them